app/assets/javascripts/components/course_creator/course_creator.jsx

```diff
@@ -310,11 +323,18 @@ const CourseCreator = createReactClass({
     const cloneOptions = showNewOrClone ? '' : ' hidden';
     const controlClass = `wizard__panel__controls ${courseDates}`;
     const selectClass = showCloneChooser ? '' : ' hidden';
-    const options = this.props.cloneableCourses.map((course, i) => <option key={i} data-id-key={course.id}>{course.title}</option>);
+    const options = this.props.cloneableCourses.map((course, i) => <option key={i} data-id-key={course.id} value={course.slug}>{course.title}</option>);
     const selectClassName = `select-container ${selectClass}`;
     const eventFormClass = this.state.showEventDates ? '' : 'hidden';
     const eventClass = `${eventFormClass}`;
-    const reuseCourseSelect = <select id="reuse-existing-course-select" ref={(dropdown) => { this.courseSelect = dropdown; }}>{options}</select>;
+    const reuseCourseSelect = <select id="reuse-existing-course-select" ref={(dropdown) => { this.courseSelect = dropdown; }} onChange={this.change}>{options}</select>;
+
+    let showCheckbox;
+    if (this.props.assignmentsWithoutUsers.length > 0) {
```

Comment:
Is `this.props.assignmentsWithoutUsers` guaranteed to always be set, so that calling `.length` on it will never throw an error?

Reply:
Just to make sure it never throws an error I've added a condition to only check `.length` if `this.props.assignmentsWithoutUsers` is present

Reply:
if there's no logical way for it to be unset, I don't think that check should be added. As along as the initial state will render without depending on a successful fetch of assignment data first, it should be fine.

Reply:
But for just a case API crashes it's better to have that check there.

Reply:
Okay. A better solution than adding the check here will be to make sure the initial state is set to an empty array or similar.

Reply:
So that is already there in global state, so i can remoive that extra check and it should not produce any error

Reply:
> So i just realised that empty array is already there in global state, so i can remoive that extra check and it should not produce any error